ui/app/components/totp-create.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+/**
+ * @module TotpCreate
+ * TotpCreate component displays the form for creating a new account
+ *
+ * @example
+ * ```js
+ * <TotpCreate
+ *  @model={{model}}
+ *  @modelForData={{@modelForData}}
+ *  @error={{this.error}}
+ *  @buttonDisabled={{this.saving}}
+ * />
+ * ```
+ * @param {object} model - the route model
+ * @param {object} modelForData - a class that helps track secret data, defined in secret-edit
+ * @param {string} error - error message to be displayed
+ * @param {boolean} buttonDisabled - if true, disables the submit button on the create/update form
+ */
+
+import Component from '@glimmer/component';
+import ControlGroupError from 'vault/lib/control-group-error';
+import Ember from 'ember';
+import keys from 'core/utils/key-codes';
+import { action, set } from '@ember/object';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { isBlank } from '@ember/utils';
+import { task, waitForEvent } from 'ember-concurrency';
+
+const LIST_ROUTE = 'vault.cluster.secrets.backend.list';
+const LIST_ROOT_ROUTE = 'vault.cluster.secrets.backend.list-root';
+const SHOW_ROUTE = 'vault.cluster.secrets.backend.show';
+
+export default class TotpCreate extends Component {
+  @tracked error = null;
+  @tracked nameWhiteSpaceWarning = false;
+  @tracked validationErrorCount = 0;
+  @tracked validationMessages = {
+    name: '',
+    url: '',
+  };
+
+  @service controlGroup;
+  @service flashMessages;
+  @service router;
+  @service store;
+
+  checkValidation(name, value) {
```

Comment:
I added the `@withModelValidations` decorator but I'm not sure where to call the validate since `totp-edit.js` extends the `role-edit.js`. Could I just add the `validate()` call in the `createOrUpdate` action or would that break for models without the decorator? https://github.com/hashicorp/vault/blob/7f0e64480fb4aec87df92359559a5a16cdeee0cd/ui/app/components/role-edit.js#L69